### DISCUSSION TITLE
v6.20.0

### DISCUSSION THREAD

**Discussion by ssbarnea** [ansible/ansible-lint#3752] [Announcements]
## Minor Changes

- Respect cgroup CPU limits when running inside containers (#3749) @ssbarnea
- Rename --write option to --fix (#3748) @ssbarnea
- Add auto-fixing implementation for `no-free-form` rule (#3725) @ajinkyau
- Add auto-fixing implementation for `partial-become` rule (#3692) @shatakshiiii
- Add auto-fixing implementation for `no-jinja-when` rule (#3721) @shatakshiiii

## Bugfixes

- Fix the detection of role vars missing role prefix (#3750) @cavcrosby
- Clarify auto-fix arguments use (#3751) @ssbarnea
- Avoid IndexError exception with jinja transform (#3747) @ssbarnea
- Fix docs typo (#3746) @ssbarnea
- Add /bin/bash executable to example correct code for pipefail rule (#3731) @Perdu
- Properly avoid `pipx` venv path to be included in altered path warning (#3730) @chrysle
- Fix transform exception with local_action with old syntax (#3743) @ssbarnea
- Support only_upgrade (#3734) @soroshsabz

<hr /><em>This discussion was created from the release <a href='https://github.com/ansible/ansible-lint/releases/tag/v6.20.0'>v6.20.0</a>.</em>

**Comment by claf** [comment]
Quick question related to the fix of role prefix detection : my role variables uses a prefix which is only a substring of the role. Can I modify this rule behavior by simply modifying var_naming_pattern as mentioned [here](https://ansible.readthedocs.io/projects/lint/rules/var-naming/) without altering other var-naming rules? For instance with : 

`var_naming_pattern: "^myprefix[a-z_][a-z0-9_]*$"`

**Reply by ssbarnea** [reply]
We need to unify this and update the documentation to include a placeholder for role name, one that would be empty for anything outside a role. This will allow us to a single configurable value that works for both cases. If you can make a PR to do this it would be great.
